From 4104f820d5c09dffda5ef72e1695a7bd5016d1b5 Mon Sep 17 00:00:00 2001
From: Thejas Nair <thejas@hortonworks.com>
Date: Mon, 22 Feb 2016 08:50:40 -0800
Subject: [PATCH 0634/1164] CDH-40043 HIVE-13093 : hive metastore does not
 exit on start failure (Thejas Nair, reviewed by
 Daniel Dai)

Change-Id: I4c6505ef19502fe9600d9a6922ce93307bd17d89
---
 .../hadoop/hive/metastore/HiveMetaStore.java       |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java b/metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java
index e5bd59f..4c35139 100644
--- a/metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java
+++ b/metastore/src/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java
@@ -6330,7 +6330,8 @@ public void run() {
         }
       }
     };
-
+    t.setDaemon(true);
+    t.setName("Metastore threads starter thread");
     t.start();
   }
 
-- 
1.7.9.5

